cypress/e2e/trafficlight/trafficlight.spec.ts

```diff
@@ -70,232 +132,70 @@ function recurse() {
-                cy.get(".mx_RightPanel_roomSummaryButton").click();
-                cy.get(".mx_RoomSummaryCard_icon_people").click();
-                cy.get(".mx_MemberList_query").type(data["userId"]);
-                cy.get(".mx_MemberList_wrapper .mx_EntityTile").click();
-                cy.get(".mx_UserInfo_verifyButton").click();
-                cy.get(".mx_UserInfo_startVerification").click();
-            } else {
-                cy.gotoAllSettings();
-                cy.get("[data-testid='settings-tab-USER_SECURITY_TAB']").click();
-                cy.contains("Verify").first().click();
-                cy.contains("Verify with another device").click();
-            }
-            return 'started_crosssign';
+            return startCrossSigning(data);
         case 'accept_crosssign':
-            // Can we please tag some buttons :)
-            // Click 'Verify' when it comes up
-            cy.get('.mx_Toast_buttons > .mx_AccessibleButton_kind_primary').click();
-            // Click to move to emoji verification
-            cy.wait(1000).then(() => {
-                // Choose whichever exists
-                Cypress.$(".mx_VerificationPanel_verifyByEmojiButton")?.trigger("click");
-                Cypress.$('.mx_VerificationPanel_QRPhase_startOption > .mx_AccessibleButton')?.trigger("click");
-            });
-            return 'accepted_crosssign';
+            return acceptCrossSigningRequest();
         case 'verify_crosssign_emoji':
-            cy.get('.mx_VerificationShowSas_buttonRow > .mx_AccessibleButton_kind_primary').click();
-            cy.get('.mx_UserInfo_container > .mx_AccessibleButton').click();
-            return 'verified_crosssign';
+            return verifyCrossSigningEmoji();
         case "verify_trusted_device":
-            cy.gotoAllSettings();
-            cy.get("[data-testid='settings-tab-USER_SECURITY_TAB']").click();
-            // For now, we only care if there are any verified devices
-            cy.contains(/^Verified devices$/);
-            cy.get(".mx_DevicesPanel_device").children();
-            return "verified";
-        case 'idle':
-            cy.wait(5000);
-            break;
-        case 'create_room':
-            cy.get('.mx_RoomListHeader_plusButton').click();
-            cy.get('.mx_ContextualMenu').contains('New room').click();
-            cy.get('.mx_CreateRoomDialog_name input').type(data['name']);
-            if (data['topic']) {
-                cy.get('.mx_CreateRoomDialog_topic input').type(data['topic']);
-            }
-            // do this to prevent https://github.com/vector-im/element-web/issues/22590, weirdly
-            // cy.get('.mx_CreateRoomDialog_name input').click();
-            // cy.wait(5000);
+            return verifyDeviceIsTrusted();
+        case "enable_dehydrated_device":
+            return enableDehydratedDevice(data);
+        case "enable_key_backup":
+            return enableKeyBackup(data);
```

Comment:
```suggestion
            const password = data["key_backup_passphrase"];
            if (!password) {
                    throw new Error("'key_backup_passphrase' not in data for action 'enable_dehydrated_device'");
            }
            return enableKeyBackup(password);
```

Perhaps this style so we can marshall/unmarshall the json here, and have just the control of the client in the method?